master/buildbot/data/builds.py

```diff
@@ -170,12 +170,17 @@ def get(self, resultSpec, kwargs):
         for b in builds:
             data = yield self.db2data(b)
             # Avoid to request DB for Build's properties if not specified
-            if filters:  # pragma: no cover
-                props = yield self.master.db.builds.getBuildProperties(b['id'])
-                filtered_properties = self._generate_filtered_properties(
-                    props, filters)
-                if filtered_properties:
-                    data['properties'] = filtered_properties
+            try:
```

Comment:
hell! no!

filters don't belong to the else: part. You may want to filter by properties even if you already filter by change (why not?)